library/src/howMuch.ts

```diff
@@ -0,0 +1,20 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as Codec from "io-ts/lib/Codec";
+import { draw } from "io-ts/lib/Decoder";
+import { matchSorter } from "match-sorter";
+import apple from "../things/apple.json";
+import { ResultObject, Thing } from "./domain";
+
+const things = Codec.array(Thing.codec).decode(apple);
```

Comment:
As discussed, it's not obvious currently how:
- we can have multiple "things" in multiple json;
- we can have "things" and other types

But it can be addressed later as it's not that obvious.

Reply:
> - we can have multiple "things" in multiple json;
Agreed, we need to think about this, maybe have a first small program to compile things sources into 1 single json that will be the input of rollup.

> - we can have "things" and other types
That can be done with tagged unions in io-ts. Something like that:
```typescript
  Codec.sum('kind')({
    thing: Codec.thing,
    trip: Codec.trip
  })
```
This tells io-ts to pick the right codec depending on the "kind" attribute.